src/Poseidon/Janno.hs

```diff
@@ -347,11 +350,59 @@ instance (ToJSON a) => ToJSON (JannoList a) where
 instance (FromJSON a) => FromJSON (JannoList a) where
     parseJSON v = JannoList <$> parseJSON v
 
+-- |A datatype to represent Relationship degree lists in a janno file
+type JannoRelationDegreeList = JannoList RelationDegree
+
+data RelationDegree = Identical | First | Second | ThirdToFifth | SixthToTenth | Unrelated | OtherDegree
+    deriving (Eq, Ord, Generic)
+
+instance ToJSON RelationDegree where
+    toEncoding = genericToEncoding defaultOptions
+
+instance FromJSON RelationDegree
+
+instance Csv.FromField RelationDegree where
+    parseField x
+        | x == "identical"    = pure Identical
+        | x == "first"        = pure First
+        | x == "second"       = pure Second
+        | x == "thirdToFifth" = pure ThirdToFifth
+        | x == "sixthToTenth" = pure SixthToTenth
+        | x == "unrelated"    = pure Unrelated -- this should be omitted in the documentation
+                                               -- relations of type "unrelated" don't have to be
+                                               -- listed explicitly
+        | x == "other"        = pure OtherDegree
+        | otherwise           = fail $ "Relation degree " ++ show x ++ 
+                                       " not in [identical, first, second, thirdToFifth, sixthToTenth, other]"
+
+instance Csv.ToField RelationDegree where
+    toField Identical         = "identical"
+    toField First             = "first"
+    toField Second            = "second"
+    toField ThirdToFifth      = "thirdToFifth"
+    toField SixthToTenth      = "sixthToTenth"
+    toField Unrelated         = "unrelated"
+    toField OtherDegree       = "other"
+
+instance Show RelationDegree where
+    show Identical            = "identical"
+    show First                = "first"
+    show Second               = "second"
+    show ThirdToFifth         = "thirdToFifth"
+    show SixthToTenth         = "sixthToTenth"
+    show Unrelated            = "unrelated"
+    show OtherDegree          = "other"
+
 -- | A data type to represent a sample/janno file row
 -- See https://github.com/poseidon-framework/poseidon2-schema/blob/master/janno_columns.tsv
 -- for more details
 data JannoRow = JannoRow
-    { jIndividualID                 :: String
+    { jPoseidonID                   :: String
```

Comment:
I think I only changed the order at one point: in the `jannoHeader` list. Because that's what's relevant for the output order. Adjusting the order anywhere else felt like a waste of time.

https://github.com/poseidon-framework/poseidon-hs/blob/d6b0871319a640c1c1c06eeb1b339402c81cc709/src/Poseidon/Janno.hs#L566-L570

Reply:
Ah - and in `survey`. There the order matters as well.

Reply:
#156